ingest/usecase/ingest_usecase.go

```diff
@@ -473,6 +478,22 @@ func processSQSModelMut(sqsModel *sqsdomain.SQSPool) error {
 		newPoolDenoms = append(newPoolDenoms, denom)
 	}
 
+	// Sort the pool denoms by balance amount
+	// This is useful for edge case handling for certain pools such as alloyed
+	// where the token amounts might get imbalanced, making the liqudiity of one denom completely zero.
+	// In that case, we would like to deprioritize the out-of-liquidity denoms.
+	sort.Slice(newPoolDenoms, func(i, j int) bool {
```

Comment:
I think I am missing some ctx on how ordering the denoms by balance helps solve the de-prioritization

Reply:
Our candidate route algorithm does BFS by goring through the pool denoms in order.

There is a limit of 20 candidate routes.

The bug was that all 20 candidate routes only container WBTC that had no liquidity.

With this idea, if liquidity changes, the denom order will change. Since cache also gets invalidated, we would recompute the routes where a higher liquidity token is first, as a result, reconstructing new 20 valid routes

Reply:
Ah, I understand now, thanks